src/Form/AdminSettingsForm.php

```diff
@@ -160,32 +167,30 @@ private function getUserRoles() {
       $values[$rid] = [
         'rid' => $rid,
         'name' => $name,
-        'weight' => $config[$rid]['weight'] ? (int) $config[$rid]['weight'] : 0,
+        'weight' => $config[$rid]['weight'] ? $config[$rid]['weight'] : 0,
         'theme' => $config[$rid]['theme'] ? $config[$rid]['theme'] : '',
       ];
     }
 
-    uasort($values, function ($a, $b) {
-      $r = $a['weight'] - $b['weight'];
-      return $r;
-    });
+    // We need to order properly all rows on initial render like.
+    // @see \Drupal\Core\Config\Entity\ConfigEntityListBuilder::load().
+    uasort($values, [
+      'Drupal\Component\Utility\SortArray',
+      'sortByWeightElement',
+    ]);
 
     return $values;
   }
 
   /**
    * Gets a list of active themes without hidden ones.
    *
-   * @return array
-   *   An array with all active themes.
+   * @return arraystringstring|array[]
+   *   An array with all compatible active themes.
    */
   private function getThemeList() {
-
     $themes_list = [];
-
-    $installed_themes = $this->themeHandler->listInfo();
-    foreach ($installed_themes as $theme) {
-      // Exclude theme defined as hidden.
+    foreach ($this->themeHandler->listInfo() as $theme) {
```

Comment:
Result of $this->themeHandler->listInfo() should be saved in a variable. The variable should be used by the foreach